packages/babel-plugin-transform-object-rest-spread/src/index.ts

```diff
@@ -539,7 +539,15 @@ export default declare((api, opts: Options) => {
           const { right } = path.node;
 
           let refName;
-          if (t.isIdentifier(right) && path.scope.hasBinding(right.name)) {
+
+          if (
+            t.isIdentifier(right) &&
+            (pureGetters ||
+              path.scope.getBinding(right.name)?.constantViolations.every(
+                // declared in loop
+                p => p.isVariableDeclarator(),
```

Comment:
Why do we need to check this?

Reply:
`for(var x of y)`
`x` will be registered as `constantViolations`.